src/openrct2/interface/InteractiveConsole.cpp

```diff
@@ -1048,24 +932,21 @@ static int32_t ConsoleCommandSet(InteractiveConsole& console, const arguments_t&
             }
             else
             {
-                auto gameAction = ClimateSetAction(ClimateType{ newClimate });
-                GameActions::Execute(&gameAction);
-
-                console.Execute("get climate");
+                ConsoleSetVariableAction<ClimateSetAction>(console, var_name, ClimateType{ newClimate });
             }
         }
-        else if (argv[0] == "game_speed" && InvalidArguments(&invalidArgs, int_valid[0]))
+        else if (var_name == "game_speed" && InvalidArguments(&invalidArgs, int_valid[0]))
         {
             gGameSpeed = std::clamp(int_val[0], 1, 8);
```

Comment:
While you're working on this, I think this should also go through the related game action.